packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -27,6 +27,8 @@ const DocumentCommentSpecification _docCommentSpec =
 
 String _codecName = 'PigeonCodec';
 
+const String _overflowClassName = '${varNamePrefix}CodecOverflow';
```

Comment:
In general for all the generators, I don't think we should be using `varNamePrefix`; this isn't a variable name. We should do something class-appropriate in each language for this name.

We could even just make up a new convention, like `PigeonCodecReserved` as the start of the class name. I would be okay with a world where if someone names a class `PigeonCodecReserved*` *and* happened to pick a value of `*` that collided with one of our handful of generated classes, it would fail.

Reply:
I mean, there really isn't a reason to have a strangely shaped prefix anyway. We can disallow any prefix. I just didn't want to block people from using `pigeon` as a prefix, since I thought that was probably fairly common. I'll adjust the prefix to something simpler and less ugly and add it to the block list. 

This will be a breaking change though.